modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -606,69 +668,83 @@ INSFVRhieChowInterpolator::getVelocity(const Moose::FV::InterpMethod m,
   mooseAssert(neighbor && this->hasBlocks(neighbor->subdomain_id()),
               "We should be on an internal face...");
 
+  // evaluate face porosity, see (18) in Hanimann 2021 or (11) in Nordlund 2016
+  const auto face_eps = epsilon(tid)(face, time);
+
   // Get pressure gradient. This is the uncorrected gradient plus a correction from cell centroid
   // values on either side of the face
-  const VectorValue<ADReal> & grad_p = p.adGradSln(fi, time);
+  const auto & grad_p = p.adGradSln(fi, time);
 
   // Get uncorrected pressure gradient. This will use the element centroid gradient if we are
   // along a boundary face
-  const VectorValue<ADReal> & unc_grad_p = p.uncorrectedAdGradSln(fi, time);
+  const auto & unc_grad_p = p.uncorrectedAdGradSln(fi, time);
+  if (_approximate_as)
+  {
+    const auto coeff = Utility::pow<2>(fi.dCNMag()) /
+                       (fi.dCNMag() * _cs * (*_rhos[tid])(face, time) + (*_mus[tid])(face, time));
```

Comment:
we should document this somewhere
where did it come from, why is it the right dimensionality etc (that latter I m lost in right now, what is the unit of a?)

D = volume / a. From this line, D should be in m^3s/kg, so a in kg/s

Reply:
from the NSFVMomentum kernel I m getting 
a = area * v * rho * derivative of v so same? derivative of v with regards to v is unitless

Reply:
I thought at the beginning that you only wanted to switch the advection contribution's velocity factor to a characteristic speed. This only has the orthogonal component in the viscous contribution. Have you tested it with nonorthogonal meshes? This might still work, I am just curious.

Reply:
If you just switched the advecting speed to a characteristic speed, you wouldnt need rho and mu on the RC object. The only thing that needs to change is the A-collection routine in the advection kernel.

Reply:
At the same time, I suppose this is simpler and if it works with nonorthogonal problems, might even be better.

Reply:
It's my opinion that all Rhie-Chow is is a stabilization technique for pressure checker-board. It's effect is to remove the checkerboard in the pressure at the cost of introducing error in the velocity solution. More Rhie-Chow, less pressure checker-board but more error in velocity. This is borne out in the the errors shown at the top of the PR. And here I show the MMS plots and errors for the finest mesh for the skewed vortex problem which shows the same.

The actual non-orthogonal correction comes through in the `INSFVMomentumDiffusion` kernel regardless of RC or not.

### Current Rhie-Chow
![skewed](https://github.com/idaholab/moose/assets/10248304/fe79a25a-f55f-4222-b0ca-9ee092c6e228)

Finest run errors:
```
Postprocessor Values:
+----------------+----------------+----------------+----------------+----------------+
| time           | L2p            | L2u            | L2v            | h              |
+----------------+----------------+----------------+----------------+----------------+
|   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |
|   1.000000e+00 |   7.334300e-04 |   5.602477e-06 |   5.758965e-06 |   2.033090e-02 |
+----------------+----------------+----------------+----------------+----------------+
```

### This PR
![skewed-approx](https://github.com/idaholab/moose/assets/10248304/2eb268ac-bc51-485b-a591-bf5a92bb2c9f)

Finest run errors:

```
Postprocessor Values:
+----------------+----------------+----------------+----------------+----------------+
| time           | L2p            | L2u            | L2v            | h              |
+----------------+----------------+----------------+----------------+----------------+
|   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |   0.000000e+00 |
|   1.000000e+00 |   4.380186e-04 |   6.682908e-06 |   6.689336e-06 |   2.033090e-02 |
+----------------+----------------+----------------+----------------+----------------+
```

Reply:
> If you just switched the advecting speed to a characteristic speed, you wouldnt need rho and mu on the RC object. The only thing that needs to change is the A-collection routine in the advection kernel.

Yes, I thought about this. The issue is that I don't know how I would sub-divide the element coefficient contributions. I have to think about
- How many faces are going to add into this elemental coefficient?
- How should I handle the advection scheme?
- What direction is the wind blowing? Should I take that into consideration?

The way it is implemented now in the user-object, it's an unambiguous "isotropic" advection addition to the face coefficient, treated analogously as diffusion.

Reply:
I see the treatment here as representing the sum of the effects considered in the computation of the Reynolds number

Reply:
> we should document this somewhere

Ok, added some documentation

Reply:
Thinking more about this, I think I will give another stab at having the implementation in `INSFVMomentumAdvection`. I would be loathe to lose the contributions to the RC coefficients from friction for example